blockchain/blockchain.go

```diff
@@ -0,0 +1,275 @@
+					pid, err := peer.Decode(strings.TrimSuffix(addr, ".invalid:80"))
+					if err != nil {
+						return nil, err
+					}
+					return gostream.Dial(ctx, h, pid, FxBlockchainProtocolID)
+				},
+			},
+		},
+		authorizedPeers: make(map[peer.ID]struct{}),
+	}
+	if bl.authorizer != "" {
+		if err := bl.SetAuth(context.Background(), h.ID(), bl.authorizer, true); err != nil {
+			return nil, err
+		}
+	}
+	return bl, nil
+}
+
+func (bl *FxBlockchain) Start(ctx context.Context) error {
+	listen, err := gostream.Listen(bl.h, FxBlockchainProtocolID)
+	if err != nil {
+		return err
+	}
+	bl.s.Handler = http.HandlerFunc(bl.serve)
+	go func() { bl.s.Serve(listen) }()
+	return nil
+}
+
+func (bl *FxBlockchain) Seeded(ctx context.Context, to interface{}, r SeededRequest) ([]byte, error) {
+	var method string
+	var addr string
+
+	switch v := to.(type) {
+	case string:
+		method = http.MethodPost
+		addr = "http://" + v + "account/" + actionSeeded
+	case peer.ID:
+		if bl.allowTransientConnection {
+			ctx = network.WithUseTransient(ctx, "fx.blockchain")
+		}
+		method = http.MethodGet
+		addr = "http://" + v.String() + ".invalid/" + actionSeeded
+	default:
+		return nil, errInvalidType
+	}
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(r); err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, addr, &buf)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := bl.c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	b, err := io.ReadAll(resp.Body)
+	switch {
+	case err != nil:
+		return nil, err
+	case resp.StatusCode != http.StatusOK:
+		return nil, fmt.Errorf("unexpected response: %d %s", resp.StatusCode, string(b))
+	default:
+		return b, nil
+	}
+}
+
+func (bl *FxBlockchain) serve(w http.ResponseWriter, r *http.Request) {
+	from, err := peer.Decode(r.RemoteAddr)
+	if err != nil {
+		log.Debugw("cannot parse remote addr as peer ID", "err", err)
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+	action := path.Base(r.URL.Path)
+	if !bl.authorized(from, action) {
+		log.Debugw("rejected unauthorized request", "from", from, "action", action)
+		http.Error(w, "", http.StatusUnauthorized)
+		return
+	}
+	switch action {
+	case actionSeeded:
+		bl.handleSeeded(from, w, r)
+	case actionAuth:
+		bl.handleAuthorization(from, w, r)
+	default:
+		http.Error(w, "", http.StatusNotFound)
+	}
+}
+
+func (bl *FxBlockchain) handleSeeded(from peer.ID, w http.ResponseWriter, r *http.Request) {
+	log := log.With("action", actionSeeded, "from", from)
+	defer r.Body.Close()
+
+	var p SeededRequest
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		log.Debugw("cannot parse request body", "err", err)
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+	go func() {
+		ctx := context.TODO()
+		if bl.allowTransientConnection {
+			ctx = network.WithUseTransient(ctx, "fx.blockchain")
+		}
```

Comment:
This is going to end up double wrapping the context with transient which will not make sense if the handling involves calling local endpoint over plain HTTP. 

See earlier comment on breaking local vs remote "seeded" request handling.